.circleci/config.yml

```diff
@@ -66,40 +66,64 @@ jobs:
         steps:
             - install-and-test
 
-    build-docker:
+    build:
         machine: true
         steps:
             - checkout
-            - run: docker run --rm -i hadolint/hadolint < Dockerfile
-            - run: docker build -t speccy:$CIRCLE_SHA1 .
+            - run: 
+                name: Lint Dockerfile
+                command: docker run --rm -i hadolint/hadolint < Dockerfile
+            - run: 
+                name: Attempt Docker Build
+                command: docker build .
 
     push-docker:
         machine: true
+        environment:
+            - DOCKER_REPOSITORY: wework/speccy
         steps:
             - checkout
-            - run: docker build -t speccy:$CIRCLE_SHA1 .
-            - run: docker login -u $DOCKER_LOGIN -p $DOCKER_PASSWORD
-            - run: docker tag speccy:$CIRCLE_SHA1 wework/speccy:nightly
-            - run: if [ "$CIRCLE_TAG" != "" ]; then docker tag speccy:$CIRCLE_SHA1 wework/speccy:latest; fi;
-            - run: if [ "$CIRCLE_TAG" != "" ]; then docker tag speccy:$CIRCLE_SHA1 wework/speccy:$CIRCLE_TAG; fi;
-            - run: docker push wework/speccy
+            - early-return-forks
+            - run: 
+                name: 
+                command: |
+                    docker_tag=$DOCKER_REPOSITORY:${CIRCLE_SHA1:0:7}
```

Comment:
this `${CIRCLE_SHA1:0:7}` has been identified as not working for 100% of shells, use `$CIRCLE_SHA1 | cut -c1-7` instead